h2o-core/src/main/java/water/util/VecUtils.java

```diff
@@ -838,4 +838,66 @@ public void map(Chunk c, NewChunk nc) {
     }
   }
 
+  /**
+   * shuffle values of a Vec {@link Vec}.
+   * 
+   * The 'new' values of the vector will be the already exisitng one's however randomly shuffeled
+   * 
+   * 
+   */
+  
+  public static class shuffleVec extends MRTask<shuffleVec>{
+    public Vec _vec;
+        
+    public shuffleVec(Vec vec) {this._vec = vec;}
+        
+        @Override
+        public void map(Chunk ch/*, NewChunk nc*/) {
+//          nc.alloc_doubles(ch._len);
+                
+            for (int i = 0; i < ch._len; ++i) {
```

Comment:
2. Here you are going through the rows of chunk and modify only `ch._len` numbers of rows in the` _vec`. But, `ch._len` is usually smaller than `_vec.len`. So you change here only **the first** `ch._len` number of rows in the `_vec`.

Reply:
So instead of passing ```Chunk ch``` I believe I should pass ```Chunk [] chs```  and go through 
```for (int col = 0; col < chs.length; ++col) {
            Chunk i_col = cs[col];
            for (int row = 0; row < i_col._len; ++row) { ... }}```
          

or is that only for Frames?